Packaging, Export Error Handling, Ignition Target & Type Dropdowns

mdbernard opened the pull request.

mdbernard committed:
Renamed chemkedGui.py to chemked-gui.py and moved to subdirectory chemked-gui

mdbernard committed:
Merge branch 'master' of https://github.com/mdbernard/ChemKED-gui

mdbernard committed:
Created __init__.py in  chemked-gui directory, created setup.py

mdbernard committed:
Renamed chemked-gui to chemked_gui

Changed all references to chemked_gui to resolve import error upon attempting to import a module with a "-" character in the name

mdbernard committed:
Reversed change from "_" to "-" in module names because that didn't work

mdbernard committed:
Moved file `dict` from global to class variable in Contents class and changed references to file to self.file

mdbernard committed:
Export Error Message & Ignition Target/Type

Added try/except to pop an error message if there is an export error. This prevents the GUI from crashing and keeps user input data intact.

Changed Ignition Target and Type objects from QLineEdit to QComboBox and added the relevant inputs listed in the PyKED schema.

mdbernard committed:
Packaging

Added ability to install and import

"import chemkedGui" results in launching the gui immediately.

bryanwweber commented:
By the way @mdbernard Can you put the issue numbers that this fixes in a comment, so they get automatically closed when this is merged? Something like "Fixes #X, #Y, #Z"

mdbernard committed:
Added functionality to launch the chemked gui from command line

Fixes #7, #8, #9

mdbernard commented:
Fixes #7, #8, #9

bryanwweber commented on chemkedGui/chemkedGui.py:
```diff
@@ -459,68 +459,71 @@ def export(self):
-                    'facility': self.file['apparatus']['facility'].text(),
-                },
-                'datapoints': datapoints
+        # if save_location[0] != '':
```
Better not to commit commented-out code 😄 Can you remove this line?

mdbernard committed:
Renamed methods to match PEP8

mdbernard committed:
Moved lots of Contents.__init__() code to specific methods

Moved each tab's setup to its own setup method in the Contents class, rather than having one large __init__ for readability.

Also changed variable and method names from camelCase to underscore_names to match PEP8.

mdbernard committed:
Changed naming scheme from chemkedGui to chemked_gui to match PEP8.

mdbernard committed:
Updated version to 0.1.2

mdbernard committed:
Added CHANGELOG.md

Added descriptions of 0.1.0, 0.1.1, and 0.1.2 releases.

mdbernard commented:
Hello Prof.,

Here are the changes to the currently open PR since our meeting:

   - Removed that unnecessary commented line you pointed out in the export
   method.
   - Renamed lots of variables and methods to snake_case to match PEP8.
   - Renamed the directory and module from chemkedGui to chemked_gui to
   match PEP8.
   - Moved lots of code from Contents.__init__() to specific setup methods
   for each tab for better readability (tab_meta_setup, etc.).
   - Added CHANGELOG.md file and filled in basic descriptions of 0.1.0,
   0.1.1, and 0.1.2.

The skip_validation is still set to True. I'll try to take a look at where
it's going wrong when changing that to False, but I probably won't get to
it until after finals. Let me know if there's anything else that ought to
be changed.

Also, I don't remember if I ever let you know that I filled in the Wiki a
bit. Right now it just explains how to use the GUI, which is hopefully
self-explanatory. Would this be better used as a place to put more in-depth
documentation of the code rather than in the comments?

Mike

bryanwweber commented:
@mdbernard The ideal place for documentation is in the code itself, and even better is docstrings for the various classes and functions. You can follow the "Google docstring style": http://www.sphinx-doc.org/en/master/ext/example_google.html#example-google which is also used by PyKED. I think the wiki is a good spot for documenting how to use the GUI though.

Do you want me to merge this as-is or do you want to fix the skip_validation here?

mdbernard commented:
I would merge as-is for now. Users shouldn't have a problem (famous last
words). I'll make a PR with the skip_validation next.

mdbernard committed:
Added self.update() when adding or removing fields.

This should fix the bug where the user would click "Remove Species" and the field would be deleted from the exported file but the GUI would not visually update the removal until changing tabs. Having trouble recreating the bug but it hasn't happened with the self.update() line so hopefully that fixed it...

mdbernard committed:
Removed self.update() lines in field addition/removal methods because they didn't do anything.

mdbernard committed:
Correct formatting in some exported fields

Major: Changed "equivalence-ratio" from [`string`] to `float`, "amount" to `float` in each species, "file-version" to `int`, "year" to `int`, and "volume" to `int`. Moved "ignition-type" out of "composition" for each datapoint.

Minor: Used for loops instead of manually adding/removing rows in some form fields.

Bug: "List index out of range" when "skip_validation" is set to False in the export method. Must be fixed.

mdbernard committed:
Update changelog
